sql/expression/autoincrement.go

```diff
@@ -8,23 +8,40 @@ import (
 	"github.com/dolthub/go-mysql-server/sql"
 )
 
-// AutoIncrement represents a literal expression (string, number, bool, ...).
+// AutoIncrement implements AUTO_INCREMENT
 type AutoIncrement struct {
-	BinaryExpression
+	UnaryExpression
 	lastInsertId *Literal
+	autoTbl      sql.AutoIncrementTable
+	autoCol      *sql.Column
 	sync.Once
 }
 
 // NewAutoIncrement creates a new AutoIncrement expression.
-func NewAutoIncrement(lastInsertId, given sql.Expression) (*AutoIncrement, error) {
-	_, ok := lastInsertId.Type().(sql.NumberType)
+func NewAutoIncrement(ctx *sql.Context, table sql.Table, given sql.Expression) (*AutoIncrement, error) {
+	autoTbl, ok := table.(sql.AutoIncrementTable)
 	if !ok {
-		return nil, errors.New("AutoIncrement must be given a number type expression")
+		return nil, errors.New("this table does not support AUTO_INCREMENT columns")
+	}
+
+	last, err := autoTbl.GetAutoIncrementValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var autoCol *sql.Column
+	for _, c := range autoTbl.Schema() {
+		if c.AutoIncrement {
+			autoCol = c
+			break
+		}
 	}
```

Comment:
Should have an error check for a table that doesn't have any auto increment columns declared